src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1SpaceBookingService.kt

```diff
@@ -168,6 +175,72 @@ class Cas1SpaceBookingService(
     success(result)
   }
 
+  @Transactional
+  fun recordDepartureForBooking(
+    premisesId: UUID,
+    bookingId: UUID,
+    cas1NewDeparture: Cas1NewDeparture,
+  ): CasResult<Cas1SpaceBookingEntity> = validatedCasResult {
+    val premises = premisesService.getApprovedPremises(premisesId)
+    if (premises == null) {
+      "$.premisesId" hasValidationError "doesNotExist"
+    }
+
+    val existingCas1SpaceBooking = cas1SpaceBookingRepository.findByIdOrNull(bookingId)
+    if (existingCas1SpaceBooking == null) {
+      "$.bookingId" hasValidationError "doesNotExist"
+    }
+
+    val departureReason = departureReasonRepository.findByIdOrNull(cas1NewDeparture.reasonId)
+    if (departureReason == null) {
+      "$.cas1NewDeparture.reasonId" hasValidationError "doesNotExist"
+    } else if (departureReason.serviceScope != "approved-premises") {
+      "$.cas1NewDeparture.reasonId" hasValidationError "incorrectDepartureReasonServiceScope"
+    }
+
+    val moveOnCategory = moveOnCategoryRepository.findByIdOrNull(cas1NewDeparture.moveOnCategoryId)
+    if (moveOnCategory == null) {
+      "$.cas1NewDeparture.moveOnCategoryId" hasValidationError "doesNotExist"
+    } else if (moveOnCategory.serviceScope != "approved-premises") {
+      "$.cas1NewDeparture.moveOnCategoryId" hasValidationError "incorrectMoveOnCategoryServiceScope"
```

Comment:
See comment above for departureReason. We can fix in the same way (copy the serviceScopeMatches function onto the MoveOnCategoryEntity)